# HG changeset patch
# User Rik <rik@octave.org>
# Date 1390442624 28800
#      Wed Jan 22 18:03:44 2014 -0800
# Node ID d7c88126ac03fbc14c0e0087b716146624b1f68d
# Parent  6165053c56b3271d0acc9b39ba0c5dc2dfd511cc
fill.m: Don't replicate cdata matrix to x,y size when it is a string.

* fill.m: Don't replicate cdata matrix to x,y size when it is a string.
Add new demo using replication of cdata.

diff --git a/scripts/plot/draw/fill.m b/scripts/plot/draw/fill.m
--- a/scripts/plot/draw/fill.m
+++ b/scripts/plot/draw/fill.m
@@ -89,17 +89,17 @@ function h = fill (varargin)
     hax = newplot (hax);
     old_nxtplt = get (hax, "nextplot");
     unwind_protect
       set (hax, "nextplot", "add");
 
       for i = 1 : length (iargs)
         cdata = varargin{iargs(i) + 2};
         ## For Matlab compatibility, replicate cdata to match size of data
-        if (iscolumn (cdata))
+        if (iscolumn (cdata) && ! ischar (cdata))
           sz = size (varargin{iargs(i)});
           if (all (sz > 1))
             cdata = repmat (cdata, [1, sz(2)]);
           endif
         endif
 
         [htmp, fail] = __patch__ (hax, varargin{iargs(i)+(0:1)}, cdata,
                                        opts{:});
@@ -156,8 +156,20 @@ endfunction
 %! t1 = (1/16:1/8:1) * 2*pi;
 %! t2 = ((1/16:1/8:1) + 1/32) * 2*pi;
 %! x1 = sin (t1) - 0.8;
 %! y1 = cos (t1);
 %! x2 = sin (t2) + 0.8;
 %! y2 = cos (t2);
 %! h = fill (x1,y1,'r', x2,y2,'g');
 
+%!demo
+%! clf;
+%! x = [0 0
+%!      1 0.5
+%!      1 0.5
+%!      0 0];
+%! y = [0 0
+%!      0 0
+%!      1 0.5
+%!      1 0.5];
+%! c = [1 2 3 4]';
+%! fill (x, y, c);
